File: lib/src/main/java/me/ycdev/android/lib/common/apps/AppsLoadFilter.java
package me.ycdev.android.lib.common.apps;

public class AppsLoadFilter {
    /**
     * Get mounted apps only (true by default).
     */
    public boolean onlyMounted = true;

    /**
     * Get enabled apps only (true by default).
     */
    public boolean onlyEnabled = true;

    /**
     * Include all system apps (true by default).
     * Note: if this config is true, {@link #includeUpdatedSysApp} will be ignored;
     * otherwise, {@link #includeUpdatedSysApp} will be checked.
     */
    public boolean includeSysApp = true;

    /**
     * Include updated system apps (true by default).
     * Note: this config will be ignored if {@link #includeSysApp} is true.
     */
    public boolean includeUpdatedSysApp = true;

    /**
     * Include myself (true by default).
     */
    public boolean includeMyself = true;
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/apps/AppsLoadFilter.kt
package me.ycdev.android.lib.common.apps

class AppsLoadFilter {
    /**
     * Get mounted apps only (true by default).
     */
    var onlyMounted = true

    /**
     * Get enabled apps only (true by default).
     */
    var onlyEnabled = true

    /**
     * Include all system apps (true by default).
     * Note: if this config is true, [.includeUpdatedSysApp] will be ignored;
     * otherwise, [.includeUpdatedSysApp] will be checked.
     */
    var includeSysApp = true

    /**
     * Include updated system apps (true by default).
     * Note: this config will be ignored if [.includeSysApp] is true.
     */
    var includeUpdatedSysApp = true

    /**
     * Include myself (true by default).
     */
    var includeMyself = true
}
